pulsar-broker/src/test/java/org/apache/pulsar/broker/web/WebServiceTest.java

```diff
@@ -349,6 +350,36 @@ public void testBrokerReady() throws Exception {
         assertEquals(res.getResponseBody(), "ok");
     }
 
+    @Test
+    public void testCompressOutputMetricsInPrometheus() throws Exception {
+        compressOutputMetricsInPrometheus = true;
+        setupEnv(true, "1.0", true, false, false, false, -1, false);
+
+        String metricsUrl = pulsar.getWebServiceAddress() + "/metrics";
+
+        @Cleanup
+        AsyncHttpClient metricsClient = new DefaultAsyncHttpClient();
+        Response metricsRes = metricsClient.prepareGet(metricsUrl).execute().get();
```

Comment:
1. I don't really understand how this actually checks if the response was GZipped compressed?
2. I'm not even sure Transfer-Encoding header is used in GzipHandler. Usually they use Content Encoding.
3. `chunked` is not encoding used but `gzip`

Reply:
```
        HttpClient client = new HttpClient();
        client.start();
        ContentResponse response = client.GET(metricsUrl);
        assertEquals(response.getHeaders().get("Content-Encoding"), "gzip");
        client.stop();
```
This code should be used to check.